src/cljs/main/broadfcui/common.cljs

```diff
@@ -225,53 +225,81 @@
-                           (.on infobox-element
+                         (let [element$ (js/$ element)
+                               button$ (js/$ (react/find-dom-node this))]
+                           (.on element$ "hide.zf.dropdown"
+                                (fn [_]
+                                  (swap! state dissoc :render-contents?)
+                                  (after-update #(this :-render-dropdown))))
+                           (.on element$
                                 "show.zf.dropdown"
                                 (fn [_]
+                                  (swap! state assoc :render-contents? true)
+                                  (after-update #(this :-render-dropdown))
                                   (.on (js/$ "body")
                                        "click.zf.dropdown"
                                        (fn [e]
                                          (when
-                                          (not
-                                           (or
-                                            (.is infobox-element (.-target e))
-                                            (pos? (.-length (.find infobox-element (.-target e))))
-                                            (.is infobox-button (.-target e))
-                                            (pos? (.-length (.find infobox-button (.-target e))))))
-                                           (.foundation infobox-element "close")
+                                           (not
+                                            (or
+                                             (.is element$ (.-target e))
+                                             (pos? (.-length (.find element$ (.-target e))))
+                                             (.is button$ (.-target e))
+                                             (pos? (.-length (.find button$ (.-target e))))))
+                                           (.foundation element$ "close")
                                            (.off (js/$ "body") "click.zf.dropdown"))))))))
                        :will-unmount
                        (fn [element]
-                         (.off (js/$ (react/find-dom-node element)))
-                         (react/unmount-component-at-node (:infobox-container @locals))
-                         (.remove (:infobox-container @locals)))})
+                         (.off (js/$ element) "show.zf.dropdown")
+                         (.off (js/$ element) "hide.zf.dropdown"))})
                 :style {:whiteSpace "normal"}}
-          (:text props)])
-        infobox-container)))
-   :render
-   (fn [{:keys [locals]}]
-     [:button {:className "button-reset" :data-toggle (:infobox-id @locals)
-               :style {:cursor "pointer" :padding "0 0.5rem"
-                       :fontSize "16px" :lineHeight "1rem"}}
-      (icons/icon {:style {:color (:link-active style/colors)}} :information)])})
+          (when (:render-contents? @state)
```

Comment:
So then, if I'm understanding all of this correctly, this allows the (otherwise static) contents to be re-rendered (per the react style) whenever the dropdown is shown?